backend/client.go

```diff
@@ -194,9 +207,32 @@ func (c *backendClient) ResumeOrchestration(ctx context.Context, id api.Instance
 //
 // [api.ErrInstanceNotFound] is returned if the specified orchestration instance doesn't exist.
 // [api.ErrNotCompleted] is returned if the specified orchestration instance is still running.
-func (c *backendClient) PurgeOrchestrationState(ctx context.Context, id api.InstanceID) error {
+func (c *backendClient) PurgeOrchestrationState(ctx context.Context, id api.InstanceID, opts ...api.PurgeOptions) error {
+	req := &protos.PurgeInstancesRequest{Request: &protos.PurgeInstancesRequest_InstanceId{InstanceId: string(id)}, Recursive: true}
+	for _, configure := range opts {
+		if err := configure(req); err != nil {
+			return fmt.Errorf("failed to configure purge request: %w", err)
+		}
+	}
+	subOrchestrationInstances := make([]api.InstanceID, 0, 10)
+	var err error
+	if req.Recursive {
+		// Fetching sub-orchestration instances
+		subOrchestrationInstances, err = GetSubOrchestrationInstances(ctx, c.be, id, true)
+		if err != nil {
+			return fmt.Errorf("failed to fetch sub-orchestration instances: %w", err)
+		}
+	}
+	// Purge parent orchestration instance
 	if err := c.be.PurgeOrchestrationState(ctx, id); err != nil {
-		return fmt.Errorf("failed to purge orchestration state: %w", err)
+		return fmt.Errorf("failed to purge orchestration instance %s: %w", id, err)
+	}
+
+	// Purge sub-orchestration instances
+	for _, iid := range subOrchestrationInstances {
+		if err := c.PurgeOrchestrationState(ctx, iid, opts...); err != nil {
```

Comment:
I see we're passing `opts...` here, but why aren't we passing them in the `PurgeOrchestrationState` call for the root orchestration?

Reply:
Modified